external-crates/move/crates/move-binary-format/src/normalized.rs

```diff
@@ -773,15 +808,27 @@ impl<S: Hash + Eq> Struct<S> {
         S: Clone,
     {
         let handle = m.datatype_handle_at(def.struct_handle);
+
+        let name = pool.intern(m.identifier_at(handle.name));
+
+        let defining_module_handle = m.module_handle_at(handle.module);
+        let defining_module_address = *m.address_identifier_at(defining_module_handle.address);
+        let defining_module_name = pool.intern(m.identifier_at(defining_module_handle.name));
+        let defining_module = ModuleId {
+            address: defining_module_address,
+            name: defining_module_name,
+        };
```

Comment:
I think this could just be `ModuleID::new(pool, m.self_id())`
But, we already generate this when we first start building the module. Thoughts about just passing it around (maybe by reference) and cloning it as needed?

Reply:
This defining ID shuffle is also how the `Datatype` forms are defined, so I think for the purpose of definitions it makes more sense to use it here as well. This ensures that a type retrieved is equal to a type made.